src/resource/notes/notes.controller.ts

```diff
@@ -0,0 +1,36 @@
+import {
+    Controller,
+    Get,
+    Post,
+    Body,
+    Patch,
+    Param,
+    UseGuards,
+    ParseIntPipe,
+} from '@nestjs/common';
+import { NotesService } from './notes.service';
+import { NoteDTO } from './dto';
+import { JwtGuard } from '../../auth/guard';
+import { editNoteDTO } from './dto/editNote.dto';
+
+@UseGuards(JwtGuard)
+@Controller('notes')
+export class NotesController {
+    constructor(private notesService: NotesService) {}
+
+    @Get()
+    fetchAllNotes(@Param('id', ParseIntPipe) userId: number) {
+        return this.notesService.fetchAllNotes(userId);
+    }
+
+    @Post(':id')
+    addNote(@Param('id', ParseIntPipe) id: number, @Body() noteDto: NoteDTO) {
+        return this.notesService.addNote(id, noteDto);
```

Comment:
We can compare id from both DTO and path param to make sure it is a proper request